pkg/baremetal/metal3remediation_manager.go

```diff
@@ -0,0 +1,434 @@
+	}
+	err = cl.Get(ctx, key, &host)
+	if apierrors.IsNotFound(err) {
+		rLog.Info("Annotated host not found", "host", hostKey)
+		return nil, err
+	} else if err != nil {
+		return nil, err
+	}
+	return &host, nil
+}
+
+// OnlineStatus returns hosts Online field value.
+func (r *RemediationManager) OnlineStatus(host *bmov1alpha1.BareMetalHost) bool {
+	return host.Spec.Online
+}
+
+// GetRemediationType return type of remediation strategy.
+func (r *RemediationManager) GetRemediationType() infrav1.RemediationType {
+	if r.Metal3Remediation.Spec.Strategy == nil {
+		return ""
+	}
+	return r.Metal3Remediation.Spec.Strategy.Type
+}
+
+// RetryLimitIsSet returns true if retryLimit is set, false if not.
+func (r *RemediationManager) RetryLimitIsSet() bool {
+	if r.Metal3Remediation.Spec.Strategy == nil {
+		return false
+	}
+	return r.Metal3Remediation.Spec.Strategy.RetryLimit > 0
+}
+
+// HasReachRetryLimit returns true if retryLimit is reached.
+func (r *RemediationManager) HasReachRetryLimit() bool {
+	if r.Metal3Remediation.Spec.Strategy == nil {
+		return false
+	}
+	return r.Metal3Remediation.Spec.Strategy.RetryLimit == r.Metal3Remediation.Status.RetryCount
+}
+
+// SetRemediationPhase setting the state of the remediation.
+func (r *RemediationManager) SetRemediationPhase(phase string) {
+	r.Log.Info("Switching remediation phase", "remediationPhase", phase)
+	r.Metal3Remediation.Status.Phase = phase
+}
+
+// GetRemediationPhase returns current status of the remediation.
+func (r *RemediationManager) GetRemediationPhase() string {
+	return r.Metal3Remediation.Status.Phase
+}
+
+// GetLastRemediatedTime returns last remediation time.
+func (r *RemediationManager) GetLastRemediatedTime() *metav1.Time {
+	return r.Metal3Remediation.Status.LastRemediated
+}
+
+// SetLastRemediationTime setting last remediation timestamp on Status.
+func (r *RemediationManager) SetLastRemediationTime(remediationTime *metav1.Time) {
+	r.Log.Info("Last remediation time", "remediationTime", remediationTime)
+	r.Metal3Remediation.Status.LastRemediated = remediationTime
+}
+
+// GetTimeout returns timeout duration from remediation request Spec.
+func (r *RemediationManager) GetTimeout() *metav1.Duration {
+	return r.Metal3Remediation.Spec.Strategy.Timeout
+}
+
+// IncreaseRetryCount increases the retry count on Status.
+func (r *RemediationManager) IncreaseRetryCount() {
+	r.Metal3Remediation.Status.RetryCount++
+}
+
+// SetOwnerRemediatedConditionNew sets MachineOwnerRemediatedCondition on CAPI machine object
+// that have failed a healthcheck.
+func (r *RemediationManager) SetOwnerRemediatedConditionNew(ctx context.Context) error {
+	machineHelper, err := patch.NewHelper(r.OCPMachine, r.Client)
+	if err != nil {
+		r.Log.Info("Unable to create patch helper for Machine")
+		return err
+	}
+	// TODO
+	//conditions.MarkFalse(r.OCPMachine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, "")
```

Comment:
Why is this commented out?

Reply:
I wasn't sure yet what to do with this condition, because it's used by CAPI for deleting the machine and not by OCP. I removed this now and updated the controller to delete the machine itself (similar to what the current annotation based remediation in machine actuator is doing)